generator/internal/round_robin_decorator_generator.cc

```diff
@@ -372,9 +372,10 @@ future<Status>
   CcPrint(R"""(
 std::shared_ptr<$stub_class_name$>
 $round_robin_class_name$::Child() {
-  std::lock_guard<std::mutex> lk(mu_);
-  auto child = children_[current_];
+  std::unique_lock<std::mutex> lk(mu_);
+  auto const& child = children_[current_];
   current_ = (current_ + 1) % children_.size();
```

Comment:
You might also consider strength reduction to ...
```
  if (++current_ == children_.size()) current_ = 0;
```

Reply:
Ditto

Reply:
I am going to go with your approach.  easier to read I think.